CMakeLists.txt

```diff
@@ -28,6 +28,14 @@ include(Bootstrap)
 include(Cppcheck)
 include(CodeCoverage)
 
+# Adds support for multithreading with the pthreads library
+# GNU gcc compiler must use explicit CXX flag for proper linker process
+FIND_PACKAGE(Threads REQUIRED)
+
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
```

Comment:
```suggestion
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
```

Reply:
Is it _required_ for whole project or just for a subset of project? Consider using `target_compile_options`. There is also `CMAKE_THREAD_LIBS_INIT`

Reply:
Yes. It'll be `required` for all project in future, when we'll use logger library

Reply:
I not think so. Right now it required only to link just logger library. We're not use pthread API